components/spi.rst

```diff
@@ -41,10 +43,91 @@ Configuration variables:
 - **clk_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin used for the clock line of the SPI bus.
 - **mosi_pin** (*Optional*, :ref:`Pin Schema <config-pin_schema>`): The pin used for the MOSI line of the SPI bus.
 - **miso_pin** (*Optional*, :ref:`Pin Schema <config-pin_schema>`): The pin used for the MISO line of the SPI bus.
-- **force_sw** (*Optional*, boolean): Whether software implementation should be used even if hardware one is available.
 - **id** (*Optional*, :ref:`config-id`): Manually specify the ID for this SPI hub if you need multiple SPI hubs.
+- **interface** (*Optional*): Controls which hardware or software SPI interface should be used.
+  Value may be one of ``any`` (default), ``software``, ``hardware``, ``spi``, ``spi2`` or ``spi3``, depending on
+  the particular chip. See discussion below.
+- **force_sw** (*Optional*, **Deprecated**, boolean): Whether software implementation should be used even if a hardware
+  interface is available. Default is ``false``.
+
+Interface selection:
+--------------------
+
+ESP32 and ESP8266 chips have several hardware SPI controller interfaces - usually the first one or two are reserved for use to access
+the flash and PSRAM memories, leaving one or two user-accessible SPI controllers. An SPI hub configured in
```

Comment:
```suggestion
the flash and PSRAM memories, leaving one or two user-accessible SPI controllers. A SPI hub configured in
```

Reply:
I pronounce "SPI" as "ess pee eye" which would require "An" before it (starts with a vowel sound.) I'll reword it to avoid the question :-)

WRT to the other comments, I'll clean them up - I'm going to use this nomenclature (avoiding "hub" which is more an internal term from what I've seen.)

* Component - the singleton SPI component, with one or more controllers;
* Controller - a hardware or software controller associated with a bus (set of data/control lines.)
* Device - a device attached to a controller via its bus.
* Interface - the controller implementation (hw or sw).

Reply:
Sounds good! Thanks again. 🍻